packages/server-wallet/src/models/channel.ts

```diff
@@ -173,6 +177,67 @@ export class Channel extends Model implements RequiredColumns {
     });
   }
 
+  public static directFundingStatus(
+    supported: SignedStateVarsWithHash | undefined,
+    fundingFn: (address: Address) => ChannelStateFunding,
+    myParticipant: Participant,
+    fundingStrategy: FundingStrategy
+  ): FundingStatus {
+    if (fundingStrategy !== 'Direct') {
+      return 'Uncategorized';
+    }
+
+    const outcome = supported?.outcome;
+    if (!supported || !outcome) {
+      return 'Uncategorized';
+    }
+
+    const {allocationItems, assetHolderAddress} = checkThat(outcome, isSimpleAllocation);
+
+    // Collapse all allocation items with my destination into one
+    const myAllocation = allocationItems
+      .filter(ai => ai.destination === myParticipant.destination)
+      .reduce(
+        (soFar, currentAi) => ({
+          ...soFar,
+          amount: BN.add(soFar.amount, currentAi.amount),
+        }),
+        {
+          destination: myParticipant.destination,
+          amount: BN.from(0),
+        }
+      );
+
+    const funding = fundingFn(assetHolderAddress);
+
+    const amountTransferredToMe = funding.transferredOut
+      .filter(tf => tf.toAddress === myAllocation.destination)
+      .reduce((soFar, currentAi) => BN.add(soFar, currentAi.amount), BN.from(0));
+    // Note that following case:
+    // - The total amount allocated to me are zero
+    // - Only one final state is signed, and that state is supported.
+    // This channel is categorized as Defunded even though all final states might not be signed yet.
+    if (supported.isFinal && BN.gte(amountTransferredToMe, myAllocation.amount)) {
+      return 'Defunded';
+    }
+
+    const fullFunding = allocationItems.map(a => a.amount).reduce(BN.add, BN.from(0));
+    if (BN.eq(amountTransferredToMe, 0) && BN.gte(funding.amount, fullFunding)) {
+      return 'Funded';
+    }
+
+    const allocationsBeforeMe = _.takeWhile(
+      allocationItems,
+      a => a.destination !== myAllocation.destination
```

Comment:
Consider this: https://github.com/statechannels/statechannels/issues/2947#issuecomment-732364811

If we do, or plan to, allow repeated destinations, do you know how to corret this?

Reply:
I think this is a good point, but should probably be tackled in a separate PR. Now that all funding logic is in one place, we can implement and test allowing repeated destinations.